```======================================================================

FAIL: test_case_creation (custom.enikshay.nikshay_datamigration.tests.test_create_enikshay_cases.TestCreateEnikshayCases)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 264, in inner

    return helper(*args, **kwargs)

  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 237, in __call__

    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)

  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 252, in call_with_settings

    fn(*args, **kwargs)

  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 233, in fn_with_pre_and_post

    self.fn(*args, **kwargs)

  File "/mnt/commcare-hq-ro/custom/enikshay/nikshay_datamigration/tests/test_create_enikshay_cases.py", line 144, in test_case_creation

    occurrence_case.indices

AssertionError: Element counts were not equal:

First has 1, Second has 0:  CommCareCaseIndex(identifier='host', referenced_type='person', referenced_id='709e579736014d719164dc0554f2d9c8', relationship='extension')

First has 0, Second has 1:  <CommCareCaseIndexSQL: CaseIndex(case_id='1d9b744f7b1c4ce5a71a6b984ba9ceba', domain='enikshay-test-domain', identifier='host', referenced_type='person', referenced_id='709e579736014d719164dc0554f2d9c8', relationship='extension)>

-------------------- >> begin captured stdout << ---------------------

<function test_case_creation at 0x7fc7a5fbe758> failed with the following settings:

settings.TESTS_SHOULD_USE_SQL_BACKEND = True```

--- a/custom/enikshay/nikshay_datamigration/tests/test_create_enikshay_cases.py
+++ b/custom/enikshay/nikshay_datamigration/tests/test_create_enikshay_cases.py
@@ -86,7 +86,6 @@ class TestCreateEnikshayCases(TestCase):
 
         super(TestCreateEnikshayCases, self).tearDown()
 
-
     @run_with_all_backends
     def test_case_creation(self):
         call_command('create_enikshay_cases', self.domain.name)
@@ -134,14 +133,15 @@ class TestCreateEnikshayCases(TestCase):
             occurrence_case.dynamic_case_properties()
         )
         self.assertEqual('Occurrence #1', occurrence_case.name)
-        self.assertItemsEqual(
-            [CommCareCaseIndex(
+        self.assertEqual(len(occurrence_case.indices), 1)
+        self._assertIndexEqual(
+            CommCareCaseIndex(
                 identifier='host',
                 referenced_type='person',
                 referenced_id=person_case.get_id,
                 relationship='extension',
-            )],
-            occurrence_case.indices
+            ),
+            occurrence_case.indices[0]
         )
 
         episode_case_ids = self.case_accessor.get_case_ids_in_domain(type='episode')
@@ -160,14 +160,15 @@ class TestCreateEnikshayCases(TestCase):
             ]),
             episode_case.dynamic_case_properties()
         )
-        self.assertItemsEqual(
-            [CommCareCaseIndex(
+        self.assertEqual(len(episode_case.indices), 1)
+        self._assertIndexEqual(
+            CommCareCaseIndex(
                 identifier='host',
                 referenced_type='occurrence',
                 referenced_id=occurrence_case.get_id,
                 relationship='extension',
-            )],
-            episode_case.indices
+            ),
+            episode_case.indices[0]
         )
 
         test_case_ids = set(self.case_accessor.get_case_ids_in_domain(type='test'))
@@ -210,14 +211,15 @@ class TestCreateEnikshayCases(TestCase):
             ]
         )
         for test_case in test_cases:
-            self.assertItemsEqual(
-                [CommCareCaseIndex(
+            self.assertEqual(len(test_case.indices), 1)
+            self._assertIndexEqual(
+                CommCareCaseIndex(
                     identifier='host',
                     referenced_type='occurrence',
                     referenced_id=occurrence_case.get_id,
                     relationship='extension',
-                )],
-                test_case.indices
+                ),
+                test_case.indices[0]
             )
 
     @run_with_all_backends
@@ -247,3 +249,9 @@ class TestCreateEnikshayCases(TestCase):
         self.assertEqual(1, len(episode_case_ids))
         episode_case = self.case_accessor.get_case(episode_case_ids[0])
         self.assertEqual(episode_case.dynamic_case_properties()['disease_classification'], 'extra_pulmonary')
+
+    def _assertIndexEqual(self, index_1, index_2):
+        self.assertEqual(index_1.identifier, index_2.identifier)
+        self.assertEqual(index_1.referenced_type, index_2.referenced_type)
+        self.assertEqual(index_1.referenced_id, index_2.referenced_id)
+        self.assertEqual(index_1.relationship, index_2.relationship)

